src/coffee/directives/api/models/parent/markers-parent-model.coffee

```diff
@@ -77,32 +81,39 @@ angular.module("uiGmapgoogle-maps.directives.api.models.parent")
             else
               @pieceMeal @scope, false
 
+          bindToTypeEvents: (typeEvents) =>
+            self = @
+            if not @origTypeEvents
+              @origTypeEvents =
+                click: typeEvents?.click
+                mouseout: typeEvents?.mouseout
+                mouseover: typeEvents?.mouseover
+            else
+              #rollback to not have stack overflow to call self over and over
+              angular.extend typeEvents, @origTypeEvents
+
+            angular.extend typeEvents,
+              click:(group) ->
+                self.maybeExecMappedEvent group, 'click'
+              mouseout:(group) ->
+                self.maybeExecMappedEvent group, 'mouseout'
+              mouseover:(group) ->
+                self.maybeExecMappedEvent group, 'mouseover'
+
+
           createAllNew: (scope) =>
             if @gManager?
               @gManager.clear()
               delete @gManager
-
-            if scope.doCluster
-              if scope.clusterEvents
-                self = @
-                if not @origClusterEvents
-                  @origClusterEvents =
-                    click: scope.clusterEvents?.click
-                    mouseout: scope.clusterEvents?.mouseout
-                    mouseover: scope.clusterEvents?.mouseover
-                else
-                  #rollback to not have stack overflow to call self over and over
-                  angular.extend scope.clusterEvents, @origClusterEvents
-
-                angular.extend scope.clusterEvents,
-                  click:(cluster) ->
-                    self.maybeExecMappedEvent cluster, 'click'
-                  mouseout:(cluster) ->
-                    self.maybeExecMappedEvent cluster, 'mouseout'
-                  mouseover:(cluster) ->
-                    self.maybeExecMappedEvent cluster, 'mouseover'
-
-              @gManager = new ClustererMarkerManager @map, undefined, scope.clusterOptions, scope.clusterEvents
+            #support backwards comapat clusterEvents and clusterOptions
```

Comment:
All in all here things are going to a more generic approach. So a little renaming here.